dali/operators/reader/coco_reader_op.cc

```diff
@@ -125,6 +125,10 @@ instance of an object is lower than this value, the object will be ignored.)code
   .AddOptionalArg("image_ids",
       R"code(If set to True, the image IDs will be produced in an extra output.)code",
       false)
+    .AddOptionalArg<vector<string>>("files", R"(A list of file paths to read the data from.
+
+If ``file_root`` is provided, the paths are treated as being relative to it.
+This argument is mutually exclusive with ``preprocessed_annotations``.)", nullptr)
```

Comment:
This documentation doesn't explain what this parameter does. I think it can and will be confused with `annotations_file` - the documentation should clearly state that this list is rally just a filter for the images present in the annotation file. If it refers just to the images, perhaps it should be called so.


Nitpick: Weird indent.

```suggestion
    .AddOptionalArg<vector<string>>("images", R"(A list of image paths..

If specified, it acts as a filter for the file paths present in the annotation file. If left unspecified or set to None, all images listed in the annotation file are read.

If ``file_root`` is provided, the paths are treated as being relative to it.
This argument is mutually exclusive with ``preprocessed_annotations``.)", nullptr)
```
Also, I think that it would be nice to extract the directory path from the location of the `annotations_file` - like is done for `file_list` in FileReader and NumpyReader.

Reply:
I'll take your suggestions except for the name of the argument ("images" instead of "files"). This argument is actually parsed by FileLoader, and it's shared with other readers that already have the "files" argument.